core/shared/src/main/scala/zio/prelude/ForEach.scala

```diff
@@ -224,6 +224,40 @@ trait ForEach[F[+_]] extends Covariant[F] { self =>
   def nonEmpty[A](fa: F[A]): Boolean =
     !isEmpty(fa)
 
+  /**
+   * Partitions the collection based on the specified function.
+   */
+  def partitionMap[A, B, C](
+    fa: F[A]
+  )(f: A => Either[B, C])(implicit both: IdentityBoth[F], either: IdentityEither[F]): (F[B], F[C]) = {
+    implicit val covariant: Covariant[F]       = self
+    implicit val leftIdentity: Identity[F[B]]  = Identity.fromIdentityEitherCovariant
+    implicit val rightIdentity: Identity[F[C]] = Identity.fromIdentityEitherCovariant
+    foldMap(fa) { a =>
+      f(a) match {
+        case Left(b)  => (b.succeed, either.none)
+        case Right(c) => (either.none, c.succeed)
+      }
+    }
+  }
+
+  /**
+   * Partitions the collection based on the specified effectual function.
+   */
+  def partitionMapM[G[+_]: IdentityFlatten: Covariant, A, B, C](
+    fa: F[A]
+  )(f: A => G[Either[B, C]])(implicit both: IdentityBoth[F], either: IdentityEither[F]): G[(F[B], F[C])] = {
```

Comment:
Could we generalise on Either to be just another M[_] that has a bifoldable. ?

Reply:
Our equivalent would be `BiForEach` which we don't have right now but potentially could. I'm not sure it is useful enough. What data types do you typically use this with other than `Either`?